sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -179,10 +194,15 @@ bool test_main(void) {
       &plic, kPlicTarget, kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
       kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
 
-  // Executing the test using a randon time between 1 and 15 ms to make sure
+  // Executing the test using a randon time between 25 and 40 ms to make sure
   // the aon timer is generating the interrupt after the choosen time and theres
-  // no error in the reference time measurement.
-  uint32_t sleep_time = rand_testutils_gen32_range(1, 15) * 1000;
+  // no error in the reference time measurement. 25ms minimum is required as the
+  // Verilator simulation has a low simulated clock rate (500 kHz) for Ibex so a
+  // single instruction takes a minimum of 2us to execute. A few hundred cycles
+  // is required for the interrupt to note the elapsed time so measured elapsed
+  // time is always ~1ms greater than the sleep time. So the test fails with an
+  // unacceptable time variance when the sleep time is too low.
+  uint32_t sleep_time = rand_testutils_gen32_range(25, 40) * 1000;
```

Comment:
I've noticed strange behaviors when I set irq times below 1ms. Now I know Why.
However there's a discussion on this other [PR](https://github.com/lowRISC/opentitan/pull/10046#discussion_r787606353) about the time this test takes to excute.
Can we live with this test taking more than 15 minutes to execute?

Reply:
@sriyerg could you please share your opinion?

Reply:
Additionally moving the `time_elapsed = tick_count_get();` after the timer configuration `aon_timer_testutils_wakeup_config(aon_timer, sleep_cycles);` seems to have resolved the intermittency for small timeouts

Reply:
Maybe the solution is to think differently about how the elapsed time test works. It's definitely a test we want to have the issue is simply you have a few hundred cycles latency between the interrupt firing and reading the current time to calculate the elapsed time. I can see two solutions:

1. Write a special interrupt handler for this test in assembly that gets called directly from the interrupt vector table. It would just read from rv_timer, stashing the result in some fixed location and then call the 'normal' interrupt handler. This should greatly the reduce the latency from the interrupt firing to the rv_timer read. We can also alter aon_timer_testutils_wakeup_config to return the time rather than reading it just before. Then we can read it just before the wakeup timer is enabled further reducing the extra time.

2. Calculate the allowed time variance in terms of cycles rather than just +- 5% of time. Don't allow elapsed time to be lower than sleep time, only higher (i.e. it's must be within [sleep_time, sleep_time + allowed_variance]).

The issue with 2 is at sufficiently small time values our allowed variance becomes a large % of the sleep time making the test a little useless. 1 is probably the better solution.

Reply:
I agree, 1 looks better. But will consume more time to implement.
I also see two extra options:

1. Use option 2 and define a minimum interrupt time.
2. If possible, keep 25ms and just run it on the FPGA?

Reply:
Use `kDeviceType` to set different timeout values for Verilator / FPGA and DV as appropriate. Can use higher timeout values for FPGA. 
15 min runtime for DV is acceptable. Prefer to not overcomplicate this.